### DISCUSSION TITLE
Per show Undesired/Preferred words

### DISCUSSION THREAD

**Discussion by mmhere** [pymedusa/Medusa#7610] [Ideas & Feature requests]
I note this from changelog:

> 0.3.0 (2019-03-13)
> New Features
> Added per show required/preferred words exclude option (#6033)

Yet when I Edit a show and go to the Advanced tab, I only see "exclude" capability for Ignored words and for Required words.

There isn't a way, for example, to say that a particular show is preferred to include "WEB" as a keyword so that, say, 1080p WEB episodes are downloaded after 1080p HDTV episodes have already been obtained.  It seems like I'd have to set WEB as a preferred keyword globally, but this may not be appropriate for all shows.

Am I missing something?

Similarly, having per-show specialization for the Undesired words would be nice (and also not have to set it globally for shows to which it does not apply).  An "exclude" option exists per show for Ignored words but not for the Undesired words.

**Comment by medariox** [comment]
If you an toggle exclude they will be excluded, otherwise they will be added.

**Comment by mmhere** [comment]
That's not what I mean tho.  The per-show lets you amend **Required** words and **Ignored** words.  These are the words that will totally kill a Snatch if conditions are not met.

It does not let you amend *Preferred* words and *Undesired* words, which let you tweak which things are Snatched, resulting in an initial snatch being overridden later by a "better match."

**Comment by mmhere** [comment]
So can we re-open this issue?  I don't want to open a new item for this issue, which could be a bit of a rudeness.

**Comment by medariox** [comment]
Oh, sorry misread there.

**Comment by mmhere** [comment]
Thanks for re-opening.  Please let me know if I misunderstand the intent behind *Undesired* and *Preferred* words.  It would simply be nice to have [on a per-show basis] preferences for say getting the WEB content without having to set this for all shows.  It seems like the change might be similar to https://github.com/pymedusa/Medusa/pull/6033 mentioned in the 0.3.0 release from last year.

Having not inspected the code (and GUI forms) myself, and being a SW dev, it of course may not be that simple... :smile:

**Comment by medariox** [comment]
What's wrong with setting WEB qualities as preferred and the other as allowed? I think that's what you are looking for? Adding even more settings really reduces the usability of the software.

**Comment by mmhere** [comment]
The question is making it per show.  Don't want to establish such 
preferences globally but only for a handful of shows.  Precedent is 
already set to specialize Required words and Ignored words per show.

Why not the same for the secondary hints that are called "Undesired" and 
"Preferred" words?

**Comment by Gitarski** [comment]
I was also looking for this new feature.

Here's another example.  Suppose your happy with your general/global required words and ignored words.  However, for a particular show, you know that with those settings, it picks up episodes that are not what you want.  Or, perhaps another download is better quality.  In the case I'm thinking about, the one that is picked looks like it is stretched video.  I know that the items from the 'ABC' group have resolution and aspect ratio without any weird or incorrect stretching.  For this particular show, I'd like to pick the items from the ABC group over others, if they are available or become available later.

So, this would be a preferred and/or ignored word(s) that is/are specified for each show, if needed.

**Comment by p0psicles** [comment]
Maybe it could be applicable for undesired words. But as of now preferred words isn't working optimal. So no new config options are going to be added there. First we need a system that can upgrade qualities using preferred. This has a very low prio to us. As there are already of others things to implement first.

**Comment by medariox** [comment]
@Gitarski 
You can already add per show ignored/required words and that would be the exact use case for it.

**Comment by Gitarski** [comment]
Yeah.  I think you're right.  It's close enough to use the ignored/required words per show.  If the show isn't found after a while with those required words, I can always manually pick another one and it would be obvious to determine which ones need to be manually handled from the main show list.

The idea for this request was to have some ranking so that an item would be downloaded and if the better/preferred item showed up, to download that.  Similar to if a PROPER were to come along later and would be downloaded.

I don't like the idea of continually determining if a different/other item should be downloaded.  If that was done for every episode, for every show and if everyone did that, it would be way too many additional requests.

So, after thinking about it some more, I agree that the current ignored/required words handling per show is actually a good/sufficient implementation for the use case being discussed.

Thank you.

**Comment by mmhere** [comment]
I respectfully disagree that per-show exceptions for the "softer" hints (Prefer, Undesired) are not useful.  Sometimes you observe over time that certain shows have quite a lot of quality levels delivered so you'd like to take advantage of that selection and get "just the right one" for you.

Yet, you may not want to make these exceptions be the case globally for all shows.  Maybe there are shows you care more about and would like to have a level of specificity for them that you don't want to set globally.

Maybe I'll think further on how to [better] word what I feel would be a useful enhancement.  I think I can make a compelling case if I set my wordsmithing brain to it earlier in the day.

Please don't close this just yet.

Is this an infrastructure issue?  I mean, the per-show Advanced config page already allows exception for the "harder" items (Require, Exclude), so could the same be provided for the others?

**Comment by Kvothe1970** [comment]
I would love to pick this up: 

Usecase would to prefer  a certain release group for example. Instead of going with quality alone, which does not work well HEVC x265 releases which are not UHD, but can be 720p 1080p etc. These should also be able to trigger a new search to replace them. This could be a per show setting, instead of via config/search/ preferred words which are for all releases.

**Comment by PuckStar** [comment]
I'd like this too. For instance I prefer to have a HDR version of a file. But if it can't find the word HDR then it's ok if it takes a non HDR file.
